plugins/catalog-backend/src/database/DatabaseManager.ts

```diff
@@ -35,182 +31,4 @@ export class DatabaseManager {
-          }
-        }
-        await DatabaseManager.logUpdateSuccess(
-          database,
-          location.id,
-          undefined,
-        );
-      } catch (error) {
-        logger.debug(
-          `Failed to refresh location id="${location.id}", ${error}`,
-        );
-        await DatabaseManager.logUpdateFailure(database, location.id, error);
-      }
-    }
-  }
-
-  private static async refreshSingleEntity(
-    database: Database,
-    locationId: string,
-    entity: Entity,
-    logger: Logger,
-  ): Promise<void> {
-    const { kind } = entity;
-    const { name, namespace } = entity.metadata || {};
-    if (!name) {
-      throw new Error('Entities without names are not yet supported');
-    }
-
-    const request: DbEntityRequest = {
-      locationId: locationId,
-      entity: entity,
-    };
-
-    logger.debug(
-      `Read entity kind="${kind}" name="${name}" namespace="${namespace}"`,
-    );
-
-    await database.transaction(async tx => {
-      const previous = await database.entity(tx, kind, name, namespace);
-      if (!previous) {
-        logger.debug(`No such entity found, adding`);
-        await database.addEntity(tx, request);
-      } else if (
-        !DatabaseManager.entitiesAreEqual(previous.entity, request.entity)
-      ) {
-        logger.debug(`Different from existing entity, updating`);
-        await database.updateEntity(tx, request);
-      } else {
-        logger.debug(`Equal to existing entity, skipping update`);
-      }
-    });
-  }
-
-  private static entitiesAreEqual(previous: Entity, next: Entity) {
-    if (
-      previous.apiVersion !== next.apiVersion ||
-      previous.kind !== next.kind ||
-      !lodash.isEqual(previous.spec, next.spec) // Accept that {} !== undefined
-    ) {
-      return false;
-    }
-
-    // Since the next annotations get merged into the previous, extract only
-    // the overlapping keys and check if their values match.
-    if (next.metadata.annotations) {
-      if (!previous.metadata.annotations) {
-        return false;
-      }
-      if (
-        !lodash.isEqual(
-          next.metadata.annotations,
-          lodash.pick(
-            previous.metadata.annotations,
-            Object.keys(next.metadata.annotations),
-          ),
-        )
-      ) {
-        return false;
-      }
-    }
-
-    const e1 = lodash.cloneDeep(previous);
-    const e2 = lodash.cloneDeep(next);
-
-    if (!e1.metadata.labels) {
-      e1.metadata.labels = {};
-    }
-    if (!e2.metadata.labels) {
-      e2.metadata.labels = {};
-    }
-
-    // Remove generated fields
-    delete e1.metadata.uid;
-    delete e1.metadata.etag;
-    delete e1.metadata.generation;
-    delete e2.metadata.uid;
-    delete e2.metadata.etag;
-    delete e2.metadata.generation;
-
-    // Remove already compared things
-    delete e1.metadata.annotations;
-    delete e1.spec;
-    delete e2.metadata.annotations;
-    delete e2.spec;
-
-    return lodash.isEqual(e1, e2);
-  }
 }
```

Comment:
Thats' nice! 😎

Reply:
yeah next step we should rename this one to like DatabaseFactory and add a method to create the inmemory sqlite database in one central place. But waiting for @Nek for that